ghcide/src/Development/IDE/Main.hs

```diff
@@ -249,8 +265,35 @@ defaultMain Arguments{..} = do
                 measureMemory logger [keys] consoleObserver valuesRef
 
             unless (null failed) (exitWith $ ExitFailure (length failed))
+        Index dir -> do
```

Comment:
I'm confused, `hiedb` already has an index command, where you can tell it to index a directory containing hie files. I guess this command does the work of generating `.hie` files from `.hs` files and also indexes them. I guess it would be worth clarifying this in the help text.

Reply:
I think it's useful to embed the command in HLS. Moreover, it ensures that the insertions in the database will be compatible with what HLS expects. 

I didn't surface the command through the opts parser though, so there is no help text.